core/src/main/java/org/fao/geonet/kernel/datamanager/base/BaseMetadataOperations.java

```diff
@@ -0,0 +1,278 @@
+package org.fao.geonet.kernel.datamanager.base;
+
+import static org.springframework.data.jpa.domain.Specifications.where;
+
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.fao.geonet.constants.Geonet;
+import org.fao.geonet.domain.OperationAllowed;
+import org.fao.geonet.domain.OperationAllowedId;
+import org.fao.geonet.domain.OperationAllowedId_;
+import org.fao.geonet.domain.Profile;
+import org.fao.geonet.domain.ReservedGroup;
+import org.fao.geonet.domain.ReservedOperation;
+import org.fao.geonet.domain.UserGroup;
+import org.fao.geonet.domain.UserGroupId;
+import org.fao.geonet.exceptions.ServiceNotAllowedEx;
+import org.fao.geonet.kernel.SvnManager;
+import org.fao.geonet.kernel.datamanager.IMetadataOperations;
+import org.fao.geonet.kernel.setting.SettingManager;
+import org.fao.geonet.kernel.setting.Settings;
+import org.fao.geonet.repository.MetadataRepository;
+import org.fao.geonet.repository.OperationAllowedRepository;
+import org.fao.geonet.repository.UserGroupRepository;
+import org.fao.geonet.repository.UserRepository;
+import org.fao.geonet.repository.specification.MetadataSpecs;
+import org.fao.geonet.repository.specification.UserGroupSpecs;
+import org.fao.geonet.repository.specification.UserSpecs;
+import org.fao.geonet.utils.Log;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.data.jpa.domain.Specification;
+
+import com.google.common.base.Optional;
+
+import jeeves.server.context.ServiceContext;
+
+public class BaseMetadataOperations implements IMetadataOperations {
+
+    @Autowired
+    private MetadataRepository metadataRepository;
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private OperationAllowedRepository opAllowedRepo;
+    @Autowired
+    private UserGroupRepository userGroupRepo;
+    @Autowired
+    @Lazy
+    private SettingManager settingManager;
+    @Autowired(required=false)
+    private SvnManager svnManager;
+
+    public void init(ServiceContext context, Boolean force) throws Exception {
+//        userRepository = context.getBean(UserRepository.class);
```

Comment:
I think `@Autowired` will not work due to multinode. Please check to use again `context.getBean`

Reply:
Fixed